### PR TITLE
feat: add configurable client.id to base settings

### PR DESCRIPTION
**Pull request by rgraber**
Allows configuring of the client id of producers and consumers so we can better track our message streams.

**Merge checklist:**
Check off if complete *or* not applicable:
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [x] Noted any: Concerns, dependencies, deadlines, tickets, testing instructions

### CODE REVIEW

**Review by dianakhuang — APPROVED** [review]

File: edx_event_bus_kafka/internal/config.py [hunk 1]
```diff
@@ -110,6 +109,15 @@ def load_common_settings() -> Optional[dict]:
             'sasl.password': secret,
         })
 
+    # .. setting_name: EVENT_BUS_KAFKA_CLIENT_ID
+    # .. setting_default: None
+    # .. setting_description: Identifier for the producing/consuming application. Useful for debugging. If not set
+    # .. Kafka will use 'rdkafka' as the identifier
+    client_id = getattr(settings, 'EVENT_BUS_APP_NAME', None)
```
Comment (robrap) [inline edx_event_bus_kafka/internal/config.py:116]: EVENT_BUS_KAFKA_CLIENT_ID?

File: edx_event_bus_kafka/internal/config.py [hunk 2]
```diff
@@ -110,6 +109,15 @@ def load_common_settings() -> Optional[dict]:
             'sasl.password': secret,
         })
 
+    # .. setting_name: EVENT_BUS_KAFKA_CLIENT_ID
+    # .. setting_default: None
+    # .. setting_description: Identifier for the producing/consuming application. Useful for debugging. If not set
+    # .. Kafka will use 'rdkafka' as the identifier
```
Comment (robrap) [inline edx_event_bus_kafka/internal/config.py:115]: Needs 1 or 2 space indent.

File: edx_event_bus_kafka/internal/tests/test_config.py [hunk 3]
```diff
@@ -50,13 +50,15 @@ def test_full(self):
                 EVENT_BUS_KAFKA_BOOTSTRAP_SERVERS='localhost:54321',
                 EVENT_BUS_KAFKA_API_KEY='some_other_key',
                 EVENT_BUS_KAFKA_API_SECRET='some_other_secret',
+                EVENT_BUS_APP_NAME='my_client_id',
```
Comment (robrap) [inline edx_event_bus_kafka/internal/tests/test_config.py:53]: Name?

File: CHANGELOG.rst [hunk 4]
```diff
@@ -14,6 +14,12 @@ Change Log
 Unreleased
 **********
 
+[5.6.0] - 2024-01-24
+********************
+Changed
+=======
+* Added configurable client.id to base configuration
```
Comment (robrap) [inline CHANGELOG.rst:21]: 1. Can you provide the setting name?
2. Here and/or in the docs, could you provide an example of how it is meant to be used?

File: edx_event_bus_kafka/internal/tests/test_config.py (hunk 3 above)
Comment (rgraber) [reply edx_event_bus_kafka/internal/tests/test_config.py:53]: What do you mean?

Comment (robrap) [reply edx_event_bus_kafka/internal/tests/test_config.py:53]: Originally, I thought this was supposed to be `EVENT_BUS_KAFKA_CLIENT_ID`.
However, an additional comment is that I think the following would make more sense:
```suggestion
                EVENT_BUS_APP_NAME='my_app_name',
```

**Pull request merged**